xcube/core/store/stores/directory.py

```diff
@@ -105,64 +105,75 @@ def get_data_store_params_schema(cls) -> JsonObjectSchema:
-        return str(TYPE_ID_DATASET), str(TYPE_ID_MULTI_LEVEL_DATASET), str(TYPE_ID_GEO_DATA_FRAME)
-
-    def get_data_ids(self, type_id: str = None) -> Iterator[Tuple[str, Optional[str]]]:
-        if type_id:
-            type_id = TypeId.parse(type_id)
-        if type_id == TYPE_ID_ANY:
-            type_id = None
-        self._assert_valid_type_id(type_id)
+    def get_type_specifiers(cls) -> Tuple[str, ...]:
+        return str(TYPE_SPECIFIER_DATASET), str(TYPE_SPECIFIER_MULTILEVEL_DATASET), str(TYPE_SPECIFIER_GEODATAFRAME)
+
+    def get_type_specifiers_for_data(self, data_id: str) -> Tuple[str, ...]:
+        self._assert_valid_data_id(data_id)
+        actual_type_specifier, _, _ = self._get_accessor_id_parts(data_id)
+        return actual_type_specifier,
+
+    def get_data_ids(self, type_specifier: str = None, include_titles: bool = True) -> Iterator[Tuple[str, Optional[str]]]:
+        if type_specifier:
+            type_specifier = TypeSpecifier.parse(type_specifier)
+        if type_specifier == TYPE_SPECIFIER_ANY:
+            type_specifier = None
+        self._assert_valid_type_specifier(type_specifier)
         # TODO: Use os.walk(), which provides a generator rather than a list
         # os.listdir does not guarantee any ordering of the entries, so
         # sort them to ensure predictable behaviour.
         for data_id in sorted(os.listdir(self._base_dir)):
             accessor_id_parts = self._get_accessor_id_parts(data_id, require=False)
             if accessor_id_parts:
-                actual_type_id, _, _ = accessor_id_parts
-                actual_type_id = TypeId.normalize(actual_type_id)
-                if type_id is None or type_id.is_compatible(actual_type_id):
+                actual_type_specifier, _, _ = accessor_id_parts
+                actual_type_specifier = TypeSpecifier.normalize(actual_type_specifier)
+                if type_specifier is None or type_specifier.is_compatible(actual_type_specifier):
                     yield data_id, None
 
-    def has_data(self, data_id: str) -> bool:
+    def has_data(self, data_id: str, type_specifier: str = None) -> bool:
         assert_given(data_id, 'data_id')
+        actual_type_specifier, _, _ = self._get_accessor_id_parts(data_id)
+        if type_specifier and not TypeSpecifier.parse(type_specifier).is_compatible(actual_type_specifier):
+            return False
         path = self._resolve_data_id_to_path(data_id)
         return os.path.exists(path)
 
-    def describe_data(self, data_id: str) -> DataDescriptor:
+    def describe_data(self, data_id: str, type_specifier: str = None) -> DataDescriptor:
         self._assert_valid_data_id(data_id)
+        actual_type_specifier, _, _ = self._get_accessor_id_parts(data_id)
+        if type_specifier and not TypeSpecifier.parse(type_specifier).is_compatible(actual_type_specifier):
+            raise ValueError(f'Data resource "{data_id}" is not available as type {type_specifier}. '
```

Comment:
This is more explicit:

```suggestion
            raise ValueError(f'Data resource "{data_id}" is not compatible with type specifier "{type_specifier}". '
```